libcpu/arm/arm926/cpuport.c

```diff
@@ -29,8 +29,14 @@
 #define ICACHE_MASK    (rt_uint32_t)(1 << 12)
 #define DCACHE_MASK    (rt_uint32_t)(1 << 2)
 
-extern void machine_reset(void);
-extern void machine_shutdown(void);
+RT_WEAK void machine_reset(void)
```

Comment:
where will use these functions? In the hw part, there are some rt_hw_cpu_reset/rt_hw_cpu_shutdown API.